### PR TITLE
fix: an Action should not mix subscription and rest annotations

### PR DESCRIPTION
**Pull request by aludwiko**
References #1669

### CODE REVIEW

**Comment by aludwiko** [comment]
ready for review, but to make it green I need to merge this one first: https://github.com/lightbend/kalix-jvm-sdk/pull/1668

**Review by efgpinto — APPROVED** [review]

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/Validations.scala [hunk 1]
```diff
@@ -142,7 +142,19 @@ object Validations {
...
+  }
+
+  private def actionValidation(component: Class[_]): Validation = {
+    val anySubscription = hasSubscription(component) || component.getMethods.toIndexedSeq.exists(hasSubscription)
+    val restMethods = component.getMethods.toIndexedSeq.filter(hasRestAnnotation)
+    when(anySubscription && restMethods.nonEmpty) {
+      Invalid(
+        errorMessage(
+          component,
+          s"An Action that subscribes should not be mixed with REST annotation, please move methods [${restMethods.map(_.getName).mkString(", ")}] to a separate Action component."))
```
Comment (octonato) [inline sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/Validations.scala:157]:
```suggestion
          s"An Action that subscribes should not be mixed with REST annotations, please move methods [${restMethods.map(_.getName).mkString(", ")}] to a separate Action component."))
```

**Review by octonato — APPROVED** [review]
LGTM, with a minor typo fix

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/Validations.scala (hunk 1 above)
Comment (octonato) [reply sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/Validations.scala:157]: we say: "a REST annotation" or "REST annotations"

**Pull request merged**